### PR TITLE
Us 161 As a user I want to be able to sign in to GitHub once at the launch of application.

### PR DESCRIPTION
**Pull request by alextadams88**
User is now prompted to log in to GitHub once at the launch of the application. Their username/password is stored in their preferences file for future use. All GitHub calls now use the user's credentials.

### CODE REVIEW

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import org.eclipse.egit.github.core.client.RequestException;
+
+import main.java.memoranda.braunfels.BraunfelsService;
+import main.java.memoranda.ui.App;
+import main.java.memoranda.ui.ExceptionDialog;
+import main.java.memoranda.util.Local;
+import main.java.memoranda.util.Configuration;
+
+@SuppressWarnings("serial")
+public class GitHubLoginDialog extends JDialog {
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Missing JavaDoc

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            errorLabel.setForeground(Color.RED);
+        }
+    }
+    
+    private boolean checkCredentials(String uname, String pass) throws RequestException, IOException{
+        BraunfelsService testService = new BraunfelsService(uname, pass);
+        return testService.testCredentials();
+    }
+    
+    public static void gitHubLogin() {
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Missing JavaDoc

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            errorLabel.setText(ex.getMessage());
+            errorLabel.setForeground(Color.RED);
+        }
+        catch (IOException ex) {
+            errorLabel.setText(ex.getMessage());
+            errorLabel.setForeground(Color.RED);
+        }
+    }
+    
+    private boolean checkCredentials(String uname, String pass) throws RequestException, IOException{
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Missing JavaDoc

File: src/main/java/memoranda/braunfels/BraunfelsServiceFactory.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package main.java.memoranda.braunfels;
+
+import main.java.memoranda.util.Configuration;
+
+public class BraunfelsServiceFactory {
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/BraunfelsServiceFactory.java]: Missing JavaDoc

File: src/main/java/memoranda/braunfels/BraunfelsService.java [hunk 5]
```diff
@@ -55,4 +60,12 @@
 	
 		return newList;
 	}
+	
+	public boolean testCredentials() throws RequestException, IOException{
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/BraunfelsService.java]: Missing JavaDoc

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+@SuppressWarnings("serial")
+public class GitHubLoginDialog extends JDialog {
+    
+    private static boolean loggedIn;
+    
+    GridBagConstraints gbc;
+    JPanel topPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+    JLabel header = new JLabel();
+    public JTextField ghUnameField = new JTextField();
+    public JTextField ghPassField = new JTextField();
```
Comment (kchoadley) [inline src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: I would recommend using the subclass `JPasswordField` instead here. Tested locally and functions fine.

File: src/main/java/memoranda/braunfels/BraunfelsService.java (hunk 5 above)
Comment (alextadams88) [reply src/main/java/memoranda/braunfels/BraunfelsService.java]: Javadocs added

File: src/main/java/memoranda/braunfels/BraunfelsServiceFactory.java (hunk 4 above)
Comment (alextadams88) [reply src/main/java/memoranda/braunfels/BraunfelsServiceFactory.java]: Javadocs added

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java (hunk 1 above)
Comment (alextadams88) [reply src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Javadocs added

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java (hunk 3 above)
Comment (alextadams88) [reply src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Javadocs added

File: src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java (hunk 2 above)
Comment (alextadams88) [reply src/main/java/memoranda/braunfels/ui/GitHubLoginDialog.java]: Javadocs added

**Comment by alextadams88** [comment]
Going ahead with the merge because requested changes have been made and Kristofer conditionally approved the merge with the changes.

**Pull request merged**